### PR TITLE
Update README.md setup steps

### PR DESCRIPTION
**Pull request by joshuay03**
Minor updates to the setup steps and requirements I thought would have been useful as I was setting up locally.

See comments for reasonings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,8 +69,8 @@ Mastodon acts as an OAuth2 provider, so 3rd party apps can use the REST and Stre
 
 - **PostgreSQL** 12+
 - **Redis** 4+
-- **Ruby** 3.1+
-- **Node.js** 16+
+- **Ruby** 3.2+
```
Comment (joshuay03) [inline README.md:72]: In line with [`.ruby-version`](https://github.com/mastodon/mastodon/blob/85fdbd0ad53837c9209acf3fb45811d5bae41cd9/.ruby-version#L1).

File: README.md [hunk 2]
```diff
@@ -69,8 +69,8 @@ Mastodon acts as an OAuth2 provider, so 3rd party apps can use the REST and Stre
...
+- **Node.js** 18+
```
Comment (joshuay03) [inline README.md:73]: In line with [`package.json`](https://github.com/mastodon/mastodon/blob/85fdbd0ad53837c9209acf3fb45811d5bae41cd9/package.json#L6).

File: README.md [hunk 3]
```diff
@@ -91,10 +91,12 @@ A **Vagrant** configuration is included for development purposes. To use it, com
 To set up **MacOS** for native development, complete the following steps:
 
 - Use a Ruby version manager to install the specified version from `.ruby-version`
+- Run `bundle` to install required gems
```
Comment (joshuay03) [inline README.md:94]: This and `yarn` might not be obvious to everyone.

File: README.md [hunk 4]
```diff
@@ -91,10 +91,12 @@ A **Vagrant** configuration is included for development purposes. To use it, com
...
 - Run `brew install postgresql@14 redis imagemagick libidn` to install required dependencies
 - Navigate to Mastodon's root directory and run `brew install nvm` then `nvm use` to use the version from `.nvmrc`
+- Run `yarn` to install required packages
 - Run `corepack enable && corepack prepare`
-- Run `bundle exec rails db:setup` (optionally prepend `RAILS_ENV=development` to target the dev environment)
+- Run `bundle exec rails db:setup`
```
Comment (joshuay03) [inline README.md:99]: The env var isn't optional:

https://github.com/mastodon/mastodon/blob/85fdbd0ad53837c9209acf3fb45811d5bae41cd9/config/boot.rb#L4-L13

File: README.md (hunk 1 above)
Comment (joshuay03) [reply README.md]: Reverted: https://github.com/mastodon/mastodon/compare/9726a61e255dd7853ed46e08c3a20d95887d338f..702e29d3879b26d53cd503d5c22684d80af5a42c

File: README.md [hunk 5]
```diff
@@ -91,10 +91,12 @@ A **Vagrant** configuration is included for development purposes. To use it, com
 To set up **MacOS** for native development, complete the following steps:
 
 - Use a Ruby version manager to install the specified version from `.ruby-version`
+- Run `bundle` to install required gems
 - Run `brew install postgresql@14 redis imagemagick libidn` to install required dependencies
 - Navigate to Mastodon's root directory and run `brew install nvm` then `nvm use` to use the version from `.nvmrc`
+- Run `yarn` to install required packages
 - Run `corepack enable && corepack prepare`
-- Run `bundle exec rails db:setup` (optionally prepend `RAILS_ENV=development` to target the dev environment)
+- Run `RAILS_ENV=development bundle exec rails db:setup`
```
Comment (joshuay03) [inline README.md:99]: The env var isn't optional:

https://github.com/mastodon/mastodon/blob/85fdbd0ad53837c9209acf3fb45811d5bae41cd9/config/boot.rb#L4-L13

**Comment by joshuay03** [comment]
@mjankowski Anything else I can do to help get this merged?

**Review by mjankowski — APPROVED** [review]

**Comment by mjankowski** [comment]
Not sure. Looks ok to me.

**Review by renchap — APPROVED** [review]

**Pull request merged**